src/util-decode-mime.c

```diff
@@ -2348,6 +2337,18 @@ static int ProcessMimeEntity(const uint8_t *buf, uint32_t len,
                 MAX_LINE_LEN);
     }
 
+    if ((state->state_flag != HEADER_READY && state->state_flag != HEADER_STARTED) ||
```

Comment:
More of a question.. Why not check state >= HEADER_STARTED?

Reply:
Same condition (or rather contrary condition) is used below at line 2353 before calling `ProcessMimeHeaders` xor `ProcessMimeBody`